File: Platform/Platform.Configuration/ApplicationConfiguration.cs
using System;
using Microsoft.Extensions.Configuration;

namespace Platform.Configuration
{
	public class ApplicationConfiguration
	{
		static ApplicationConfiguration()
		{
			Configuration = new ConfigurationBuilder().AddJsonFile("appsettings.json").Build();

			var environmentConnectionString = Environment.GetEnvironmentVariable("CONN_STRING");
			ConnectionString = !string.IsNullOrEmpty(environmentConnectionString)
				? environmentConnectionString
				: Database["ConnectionString"];
		}
		
		public static IConfigurationRoot Configuration { get; private set; }

		public static IConfigurationSection Database => Configuration.GetSection("Database");

		public static string ConnectionString { get; private set; }
	}
}

File: Platform/Platform.Fodels/ApplicationConfiguration.fs
namespace Platform.Fodels

open Microsoft.Extensions.Configuration
open System

type ApplicationConfiguration() =
    let envrConString = Environment.GetEnvironmentVariable("CONN_STRING")
    let envr = System.String.IsNullOrEmpty(envrConString)
    member __.Configuration = (new ConfigurationBuilder()).AddJsonFile("appconfig.json").Build()
    member __.Database = __.Configuration.GetSection("Database")
    member __.ConnectionString =
        if envr then __.Database.Item("ConnectionString")
        else Environment.GetEnvironmentVariable("CONN_STRING")
